module_4/gifts/controller/src/main/java/com/epam/esm/filter/ExceptionHandlerFilter.java

```diff
@@ -0,0 +1,33 @@
+package com.epam.esm.filter;
+
+import java.io.IOException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+
+@Component
+public class ExceptionHandlerFilter extends OncePerRequestFilter {
```

Comment:
@LizavetaSinitsyna What is the reason of using this filter?

Reply:
@AliaksandrLiulis, I use this filter in order to catch exceptions which occured during execution of the next filters (ex., filter for token validation and registration) and handle them with help of the GlobalExceptionHandler.class. If we remove this filter, the spring will return its own default response when the exeption happens, not custom responses which I created in GlobalExceptionHandler.class.

Reply:
But it starange to use filters for catching exceptions.